skflow/monitors.py

```diff
@@ -0,0 +1,157 @@
+        self.sess = sess
+        self.loss_expression_tensor = loss_expression_tensor
+        self._set_last_loss_seen()
+        if self.last_loss_seen < self.min_loss:
+            self.min_loss = training_loss
+            self.min_loss_i = self.steps
+        self._set_epoch(feed_params_fn)
+        self.report()
+
+    def _set_last_loss_seen(self):
+        """Sets last_loss_seen attribute to most recent training error"""
+        self.last_loss_seen = self.all_train_loss_buffer[-1]
+
+    def report(self):
+        """Checks whether to report, and prints loss information if appropriate"""
+        if self.verbose and (self.steps % self.print_steps == 0):
+            self._set_training_summary()
+            print(self._summary_str)
+
+    def monitor_inducing_stop(self):
+        """Returns True if the monitor requests the model stop (e.g. for early stopping)"""
+        stop_now = (self.steps - self.min_loss_i >= self.early_stopping_rounds)
+        if stop_now:
+            sys.stderr.write("Stopping. Best step:\n step {} with loss {}\n"
+                             .format(self.min_loss_i, self.min_loss))
+        return stop_now
+
+    def create_val_feed_dict(self, inp, out):
+        """Validation requires access to TensorFlow placeholders. Not used in this Monitor"""
+        pass
+
+    def _set_epoch(self, feed_params_fn):
+        """Sets self.epoch from a function that generates a dictionary including this info"""
+        if feed_params_fn:
+            feed_params = feed_params_fn()
+            self.epoch = feed_params['epoch'] if 'epoch' in feed_params else None
+
+    def _set_training_summary(self):
+        """Returns the string to be written describing training progress"""
+        avg_train_loss = np.mean(self.print_train_loss_buffer)
+        self.print_train_loss_buffer = []
+        if self.epoch:
+            self._summary_str = ("Step #{step}, epoch #{epoch}, avg. train loss: {loss:.5f}"
+                                 .format(step=self.steps, loss=avg_train_loss,
+                                         epoch=self.epoch))
+        else:
+            self._summary_str = ("Step #{step}, avg. train loss: {loss:.5f}"
+                                 .format(step=self.global_step,
+                                         loss=avg_train_loss))
+        self._modify_summary_string()
+
+    def _modify_summary_string(self):
+        """Makes monitor specific changes to printed summary. Nothing interesting in BaseMonitor"""
+        pass
+
+
+class ValidationMonitor(BaseMonitor):
+    """Monitor that reports score for validation data and uses validation data for early stopping
+
+        val_X: Validation features
+        val_y: Validation labels
+        n_classes: Number of labels in output. 0 for regression
+        See BaseMonitor for arguments
+    """
+    def __init__(self, val_X, val_y, n_classes=0, *args, **kwargs):
+        super(ValidationMonitor, self).__init__()
```

Comment:
Thanks. Missing that was a bug. I meant to pass those args to the super class. Will include in next commit.